docs/bsp.md

```diff
@@ -913,6 +914,36 @@ during compilation before completing the response.
 The client will get a `originId` field in `RunResult` if the `originId` field in the
 `RunParams` is defined.
 
+### Clean Cache Request
+
+The clean cache request is sent from the client to the server to remove cached artifacts that are associated with a given build target.
+An example use-case for the clean cache request is to free up disk space or troubleshoot build problems.
+
+After a successful clean cache request, it is expected that cached artifacts created by invoking other requests (compile, run, test) on the build target have been removed and subsequent requests result in a full recompilation.
```

Comment:
Can we tone down semantics of the `clean` operation in this paragraph? In particular, the part about "cached artifacts" (which sound like cached files on a hard drive) and the previous part about example use cases of `clean`.
  
I'm afraid the current wording can be misread by people to mean the client can expect "free disk space" to be a side effect of the operation, which makes it look like the server is forced to clean caches on disk. In bloop, a `clean` operation cleans only the incremental compiler state by passing a fresh analysis to `compile`, so we could not comply with the current wording.
  
What do you think about changing the description of this request with the following paragraph? I've tried to remove the part of example use cases and be more clear about what a user can expect (in particular, I liked the way you put it with the full recompilation expectation).
  
> The clean cache request is sent from the client to the server to reset any state associated with a given build target. The state can live either in the build tool or in the file system.
>
> The build tool defines the exact semantics of the clean cache request:
>   1. Stateless build tools are free to ignore the request and respond with a successful response.
>   2. Stateful build tools must ensure that invoking compilation on a target that has been cleaned results in a full compilation.

Reply:
That is a good formulation, I updated the description to use it unchanged.